cmd/create/cluster/cmd.go

```diff
@@ -2088,6 +2102,78 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// If OCP version >= 4.10 only
+	var machinePoolRootDisk *ocm.Volume
+
+	// Setting the machine pool root disk size is only supported on gp3 volumes
+	gp3SupportedVersion := "4.10.0"
+
+	// As of 4.10, GP3 is the default whenever possible, so this is sufficient to implement the
+	// disk size on GP3 only
+	// The only exception is when the availability zone is either local or wavelength, however this
+	// situation does not exist since LocalZone is only available as of 4.12
+	// OCM already has the fallback code to use gp2 if gp3 is not supported on a LocalZone
+	isGP3Supported, err := ocm.IsGreaterThanOrEqual(version, gp3SupportedVersion)
+	if err != nil {
+		r.Reporter.Errorf("Failed to check if the OCP version is greater than or equal to %s: %v",
+			gp3SupportedVersion, err)
+		os.Exit(1)
+	}
+
+	// If the user sets the machine pool root disk size and the cluster is not 4.10 or greater, we
+	// print a warning and ignore the value
+	if !isGP3Supported && args.machinePoolRootDiskSize != "" {
+		r.Reporter.Warnf("Setting the machine pool root disk size is only supported on OpenShift " +
+			"4.10 or greater. Ignoring the value.")
+	}
+
+	// Allow non -GP3 volume if the AZ is local or wavelength
```

Comment:
As far as I know, creating a cluster isn't supported in local or wavelength zone, is that capability recently added?

Reply:
Oh, this is just for machine pool then, right?

Reply:
Yes, you can create a machine pool in a local zone.
We have a constraint for the volume type when creating a machine pool https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/pkg/osd/machine_pool_service.go#L354.

Another check is like the one you added here, depending on the OCP version https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/pkg/models/clusters.go#L900

Reply:
The server has this validation so I removed this code.